src/main/java/org/assertj/core/api/ComparatorFactory.java

```diff
@@ -12,21 +12,87 @@
  */
 package org.assertj.core.api;
 
-import static java.lang.Math.abs;
+import org.assertj.core.internal.Doubles;
+import org.assertj.core.internal.Floats;
+import org.assertj.core.internal.RealNumbers;
 
+import java.math.BigDecimal;
 import java.util.Comparator;
+import java.util.Objects;
+
+import static java.lang.Math.abs;
 
 public class ComparatorFactory {
 
   public static final ComparatorFactory INSTANCE = new ComparatorFactory();
 
+  /**
+   * Judge whether the input number is Nan or Infinity
+   *
+   * @param number Object of Float or Double class
+   * @param <T>    type of expected and precision, which should be the subclass of java.lang.Number and java.lang.Comparable
+   * @return whether the object is Nan or Infinity
+   * @throws java.lang.NullPointerException if number is null
+   */
+  private static <T extends Number & Comparable<T>> boolean isNanOrInfinity(T number) {
```

Comment:
I thought that since T is a generic type, the judgement depends on the type of T. If I remove this method, can I modify the public compare method like this?
```java
    public int compare(Double o1, Double o2) {
        if(isNanOrInfinity(precision)){
          throw new IllegalArgumentException("Precision should not be Nan or Infinity!");
        }

        if(Doubles.instance().isNanOrInfinite(o1) || Doubles.instance().isNanOrInfinite(o2)){
          return Double.compare(o1, o2);
        }
        if (compareAsBigDecimalWithPrecision(o1, o2, precision)) {
          return 0;
        }
        return Double.compare(o1, o2);
      }

```

Reply:
Or is it a better idea to define a field so that the duplicated call to `Doubles.instance()` can be avoided?

Reply:
I have removed the `isNanOrInfinite` method in the class `ComparatorFactory` and replace it as the call of `Doubles.instance().isNanOrInfinite`.